Read database config from environment variable

joergschiller opened the pull request:
**Replace this text with a summary of the changes in your PR.
The more detailed you are, the better.**

-----------------

Before submitting the PR make sure the following are checked:

* [ ] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [ ] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [ ] Feature branch is up-to-date with `master` (if not - rebase it).
* [ ] Squashed related commits together.
* [ ] Added tests.
* [ ] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [ ] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop-rails/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.
* [ ] If this is a new cop, consider making a corresponding update to the [Rails Style Guide](https://github.com/rubocop/rails-style-guide).

[1]: https://chris.beams.io/posts/git-commit/

joergschiller commented on lib/rubocop/cop/rails/bulk_change_table.rb:
```diff
@@ -207,6 +207,18 @@ def database_yaml
+          case url
+          when %r{\Amysql2://}
+            MYSQL
+          when %r{\Apostgres(ql)?://}
```
Rails normalizes `postgres` to `postgresql` in https://github.com/rails/rails/blob/f517eefe1450b900cf3e44b8e6830d917f1530c0/activerecord/lib/active_record/database_configurations/connection_url_resolver.rb#L29

joergschiller commented:
Something I found while trying to debug the difference in RuboCop offenses between our CI and the local development environment:

https://gitlab.com/gitlab-org/customers-gitlab-com/-/issues/1029#note_499041793

koic commented:
@joergschiller Looks good to me. Can you squash your commits into one?

koic commented on lib/rubocop/cop/rails/bulk_change_table.rb:
```diff
@@ -207,6 +207,18 @@ def database_yaml
         end
 
+        def database_from_env
+          url = ENV['DATABASE_URL'].presence
```
It would be good to document of the cop this `ENV['DATABASE_URL']` behavior.

joergschiller commented:
@koic thanks for considering the changes. I've added a short description about the behaviour to the cop documentation. I also rebased on master and squashed the commits.

koic commented:
Thanks!
